### PR TITLE
Use latest watchman

### LINKED ISSUE #21545: Watchman isn't used after latest metro upgrade
# Bug Report

## Problem 
 
Watchman was locked at an older version due to [this issue](https://github.com/status-im/status-mobile/issues/16341). After the latest upgrade though, metro requires the latest watchman version and falls back to the "node watcher" (slower and buggier). This ends up (for me) with random timeouts when starting metro, since it takes the node watcher quite a bit of time to go through all the files, and we have a lot of them.

Need to make sure the issue from the latest Watchman, which was mentioned above, doesn't appear for anyone.

#### Expected behavior

Metro uses Watchman instead of the Node watcher

#### Actual behavior

Metro uses the slow and buggy Node watcher

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by clauxx**
### Summary

fixes #21545

Watchman was locked at an older version due to https://github.com/status-im/status-mobile/issues/16341. After the latest upgrade though, metro requires the latest watchman version and falls back to the "node watcher" (slower and buggier). This ends up (for me) with random timeouts when starting metro, since it takes the node watcher quite a bit of time to go through all the files, and we have a lot of them.

Need to make sure the issue from the latest Watchman, which was mentioned above, doesn't appear for anyone.

Also, I started getting a weird issue with metro's node process not dying after getting closed/killed. Sometimes it lost connection to the device and neither CTRL-C nor SIGKILL could get rid of the rogue `node` process. Inspecting the process' strace, I noticed that it tried to access non-existent files/directories inside `node-modules`, right at the moment node became non-responsive. Not sure how that's related to the unkillable process, but re-creating the `yarn.lock` seems to have fixed it (for now).

The strace:
```sh
statx(AT_FDCWD, "/home/cl/dev/work/status-mobile/node_modules", AT_STATX_SYNC_AS_STAT, STATX_ALL, {stx_mask=STATX_ALL|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFDIR|0700, stx_size=32768, ...}) = 0
statx(AT_FDCWD, "/home/cl/dev/work/status-mobile/node_modules/@react-native-community/cli-tools", AT_STATX_SYNC_AS_STAT, STATX_ALL, {stx_mask=STATX_ALL|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFDIR|0755, stx_size=4096, ...}) = 0
statx(AT_FDCWD, "/home/cl/dev/work/status-mobile/node_modules/@react-native-community/cli-tools.js", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7fffa7794d20) = -1 ENOENT (No such file or directory)
statx(AT_FDCWD, "/home/cl/dev/work/status-mobile/node_modules/@react-native-community/cli-tools.json", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7fffa7794d20) = -1 ENOENT (No such file or directory)
statx(AT_FDCWD, "/home/cl/dev/work/status-mobile/node_modules/@react-native-community/cli-tools.node", AT_STATX_SYNC_AS_STAT, STATX_ALL, 0x7fffa7794d20) = -1 ENOENT (No such file or directory)
statx(AT_FDCWD, "/home/cl/dev/work/status-mobile/node_modules/@react-native-community/cli-tools/build/index.js", AT_STATX_SYNC_AS_STAT, STATX_ALL, {stx_mask=STATX_ALL|STATX_MNT_ID, stx_attributes=0, stx_mode=S_IFREG|0644, stx_size=6722, ...}) = 0
write(26, "\33[33m\33[1mwarn\33[22m\33[39m \33[0mNo a"..., 181) = 181
write(23, "\33[36m\33[1minfo\33[22m\33[39m \33[0mRelo"..., 62) = 62
```

status: ready

### CODE REVIEW

**Comment by clauxx** [comment]
@vkjr @smohamedjavid can you test if you still get the watchman bug from https://github.com/status-im/status-mobile/issues/16341 on this branch?

**Comment by siddarthkay** [comment]
Looks good to me, metro didn't crash for me locally.
Podfile changes are missing though, I'll push.
CI failures aren't related to your code. I'll re-run the pipelines.

**Comment by siddarthkay** [comment]
component tests do fail though : `Test Suites: 18 failed, 106 passed, 124 total`
https://ci.status.im/blue/organizations/jenkins/status-mobile%2Fprs%2Ftests/detail/PR-21546/3/pipeline/29

**Comment by siddarthkay** [comment]
@clauxx : component tests pass for me locally after I revert your changes to `yarn.lock` file

**Review by siddarthkay — APPROVED** [review]

**Comment by siddarthkay** [comment]
I tested this change on Android and iOS and it works well locally.

Screenshot 2024-11-01 at 3 06 06 PM

**Comment by smohamedjavid** [comment]
> @vkjr @smohamedjavid can you test if you still get the watchman bug from #16341 on this branch?

Can confirm that the bug is NOT present anymore, and file changes (hot reload) work well as expected. Thanks 🙏

**Review by smohamedjavid — APPROVED** [review]

**Review by vkjr — APPROVED** [review]

File: .watchmanconfig [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
     "ignore_dirs": [
         "android/build",
-        "target"
+        "target",
+	".clj-kondo",
+	".shadow-cljs"
     ]
```
Comment (clauxx) [inline .watchmanconfig:7]: while debugging metro, found that watchman is keeping track of these two directories (e.g. `.clj-kondo` has ~12000 files usually and `.shadow-cljs` ~3000). On top of that, working in the editor/IDE causes constant changes to `.clj-kondo`'s cache, which re-trigger watchman making it do extra work for no reason (and also hogging up my CPU at times).

**Review by jakubgs — COMMENTED** [review]
Why not just upgrade the on in `nix/pkgs`?

**Comment by clauxx** [comment]
> Why not just upgrade the on in `nix/pkgs`?

@jakubgs do you mean upgrade watchman from the nix/pkgs file? Could do, but I'm not as versed in nix, so it seemed easier & less time consuming to just use the latest version by removing it tbh. 

On a sidenote, I removed the yarn.lock/podfile changes as it seems like it wasn't necessary for my issue. It seems like the problem with the un-killable node process was the linux kernel, where the node process was stuck in the "uninterruptible sleep" state, so that should explain why SIGKILL didn't do much. Updating the kernel version fixed the issue so far.

cc: @siddarthkay

**Review by jakubgs — COMMENTED** [review]
I think we used Nix for the package because - as the README states - we simply wanted to downgrade.

What I don't get is why is the `fb-watchman` in `yarn.lock` even older than the `4.9.0` that we provide via Nix right now?
https://github.com/status-im/status-mobile/blob/ca22b67d0e82a3cb123be2174b3ef2157e676e08/yarn.lock#L6053-L6058

**Comment by clauxx** [comment]
> I think we used Nix for the package because - as the README states - we simply wanted to downgrade.
> 
> What I don't get is why is the `fb-watchman` in `yarn.lock` even older than the `4.9.0` that we provide via Nix right now?
> 
> https://github.com/status-im/status-mobile/blob/ca22b67d0e82a3cb123be2174b3ef2157e676e08/yarn.lock#L6053-L6058

@jakubgs idk, seems like jest depends on the old version for some reason

```
**=> Found "fb-watchman@2.0.1"
info Reasons this module exists
   - "jest#@jest#core#jest-haste-map" depends on it
   - Hoisted from "jest#@jest#core#jest-haste-map#fb-watchman"
   - Hoisted from "@react-native#metro-config#metro-config#metro#metro-file-map#fb-watchman"**
```

**Comment by clauxx** [comment]
@siddarthkay @jakubgs is this good to merge? Also do we need manual qa on this?

**Comment by siddarthkay** [comment]
@clauxx : to me this PR seems good enough to merge.

**Pull request merged**